plugins/vpc-bridge/network/bridge_windows.go

```diff
@@ -75,62 +82,79 @@ func (nb *BridgeBuilder) FindOrCreateNetwork(nw *Network) error {
 
 	// HNS API does not support creating virtual switches in compartments other than the host's.
 	if nw.BridgeNetNSPath != "" {
-		return fmt.Errorf("Bridge must be in host network namespace on Windows")
+		return fmt.Errorf("bridge must be in host network namespace on Windows")
 	}
 
 	// Check if the network already exists.
 	networkName := nb.generateHNSNetworkName(nw)
-	hnsNetwork, err := hcsshim.GetHNSNetworkByName(networkName)
+	hnsNetwork, err := hcn.GetNetworkByName(networkName)
 	if err == nil {
 		log.Infof("Found existing HNS network %s.", networkName)
+		nw.NetworkID = hnsNetwork.Id
 		return nil
 	}
 
 	// Initialize the HNS network.
-	hnsNetwork = &hcsshim.HNSNetwork{
-		Name:               networkName,
-		Type:               hnsL2Bridge,
-		NetworkAdapterName: nw.SharedENI.GetLinkName(),
-
-		Subnets: []hcsshim.Subnet{
+	hnsNetwork = &hcn.HostComputeNetwork{
+		Name: networkName,
+		Type: hnsL2Bridge,
+		Ipams: []hcn.Ipam{
 			{
-				AddressPrefix:  vpc.GetSubnetPrefix(&nw.ENIIPAddresses[0]).String(),
-				GatewayAddress: nw.GatewayIPAddress.String(),
+				Subnets: []hcn.Subnet{
+					{
+						IpAddressPrefix: vpc.GetSubnetPrefix(&nw.ENIIPAddresses[0]).String(),
+						Routes: []hcn.Route{
+							{
+								NextHop:           nw.GatewayIPAddress.String(),
+								DestinationPrefix: defaultRoute,
+							},
+						},
+					},
+				},
 			},
 		},
+		SchemaVersion: hcn.SchemaVersion{
+			Major: hcnSchemaMajorVersion,
+			Minor: hcnSchemaMinorVersion,
+		},
 	}
 
-	buf, err := json.Marshal(hnsNetwork)
+	// Create the network policy for the adapter to which the vSwitch binds.
+	err = nb.addNetworkPolicy(
+		hnsNetwork,
+		hcn.NetAdapterName,
+		hcn.NetAdapterNameNetworkPolicySetting{
+			NetworkAdapterName: nw.SharedENI.GetLinkName(),
+		})
 	if err != nil {
 		return err
 	}
-	hnsRequest := string(buf)
 
 	// Create the HNS network.
-	log.Infof("Creating HNS network: %+v", hnsRequest)
-	hnsResponse, err := hcsshim.HNSNetworkRequest("POST", "", hnsRequest)
+	log.Infof("Creating HNS network: %+v", hnsNetwork)
+	hnsResponse, err := hnsNetwork.Create()
```

Comment:
Switching to HCN (v2) is a major change. There may be various behavioral differences between v1 and v2 that are hard to notice just by reading this CR. This is just a reminder to ensure proper test coverage for all ECS and EKS supported OS versions and scenarios before merging.